domain/access/state/permission.go

```diff
@@ -353,13 +472,16 @@ func (st *PermissionState) ReadAllUserAccessForTarget(ctx context.Context, targe
 		return nil, errors.Trace(err)
 	}
 
-	userAccess := make([]corepermission.UserAccess, len(permissions))
-	for i, p := range permissions {
-		user, ok := users[p.GrantTo]
-		if !ok {
-			return userAccess, errors.Annotatef(accesserrors.UserNotFound, "%q", p.GrantTo)
+	var userAccess []corepermission.UserAccess
+	for i, user := range users {
+		if user.Name == corepermission.EveryoneTagName {
+			continue
 		}
-		userAccess[i] = p.toUserAccess(user)
+		ua, err := st.generateUserAccess(user, permissions[i], dbPermission(everyoneExternals[i]))
```

Comment:
We should validate that all slices are the same length before entering this loop.

Reply:
Added a check, though SQLair also guarantees that this is the case.